botocove/cove_host_account.py

```diff
@@ -60,7 +48,16 @@ def __init__(
             self.target_regions = regions
 
         self.provided_ignore_ids = ignore_ids
-        self.target_accounts = self._resolve_target_accounts(target_ids)
+        self.provided_target_ids = target_ids
+
+        if not org_master and self.provided_target_ids:
+            account_map = _map_non_org_accounts(self.provided_target_ids)
+        else:
+            self.org_client = self._get_boto3_org_client(assuming_session)
```

Comment:
This change does break the behaviour if someone _really_ wants to use an org account but not enumerate all accounts:
"As a niche use case, I want to use my org master, provide a specific list of accounts, but describe them individually not grab the whole org list [???]"

I'm absolutely happy to drop that support and release as 1.7.0, just want to call it out!

With the thought of just dropping the org_master bool as a supported kwarg, we could just try/except this behaviour - try and list all org accounts, if it fails fallback to the degenerate dict? 

We have these cases:
Running in org master, provides list of accounts [Will now get contents of prefetched accounts, overfetches data instead of describe_acc per acc - this seems fine]
Running in org master, does not provide accounts [normal Cove usage]
Running in standalone, provides list of accounts [won't get description - no change]
Running in standalone, provides list of OUs [will fail ungracefully? We can just log an explicit error and fail here]
Running in standalone, does not provide accounts [won't work at all]

Reply:
I remember vaguely taking advantage of the behavior of the "niche" case to see a perceived speed up or increase in reliability with earlier versions of botocove (I started using it with version 1.2). I think by removing the DescribeAccount bottleneck we also remove the practical need for this use case. So it would be nice to tidy that up in version 1.7!

I like the idea of just trying to get the organization accounts and falling back to a degenerate list on some sort of error. Can we call it "graceful degredation"? :-)

If we did it like that, what would "org master" and "standalone" mean in this case? It sounds like we would assume there is an organization to query until an error is raised that proves otherwise.

Reply:
Yeah, as per https://github.com/connelldave/botocove/pull/41#discussion_r936062666 we can just drop that terminology - either we enrich data because we can hit that API, or we gracefully degrade. If we try and then list OU contents without the ability to do that, we ungracefully explode and explain why the user is doing it wrong :)

We can even allow the super-niche case of "I don't want to describe all my accounts, here's some account and OU IDs in my org master" by just suggesting an IAM policy that denies the list org accounts, mimicing that degraded behaviour while still able to run.

Reply:
> We have these cases:
> Running in org master, provides list of accounts [Will now get contents of prefetched accounts, overfetches data instead of describe_acc per acc - this seems fine]
> Running in org master, does not provide accounts [normal Cove usage]
> Running in standalone, provides list of accounts [won't get description - no change]
> Running in standalone, provides list of OUs [will fail ungracefully? We can just log an explicit error and fail here]
> Running in standalone, does not provide accounts [won't work at all]

[61f4070](https://github.com/connelldave/botocove/pull/41/commits/61f4070327c8e60d44269ce30bc97a4a7a311907) deprecates org_master and now cove beahves as if it were set True.

Surprisingly, all the tests still pass. That tells me we need new tests!

I think these cases would be good to implement as a new set of test cases. I'll work on it when I find the time. :-)